core/server/master/src/main/java/alluxio/master/metastore/caching/CachingInodeStore.java

```diff
@@ -680,9 +686,9 @@ public void removeEdge(Edge edge) {
       }
       if (entry == null || !createdNewEntry.get()) {
         // Skip caching if the cache is full or someone else is already caching.
-        return mEdgeCache.getChildIds(inodeId).values();
+        return mEdgeCache.getChildIds(inodeId, option).values();
```

Comment:
If we want to skip caching in `getChildIds`, we should add the 
```
if (entry == null || !createdNewEntry.get() || option.shouldSkipCache()) {
return mEdgeCache.getChildIds(inodeId, option).values();
}
```
so that we don't walk to the `loadChildren` caching place.